--- configure.orig	2016-08-24 22:21:08.524949225 +0200
+++ configure	2016-08-24 22:30:04.047845691 +0200
@@ -18108,7 +18108,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ax_cv_check_cflags___Werror" >&5
 $as_echo "$ax_cv_check_cflags___Werror" >&6; }
 if test x"$ax_cv_check_cflags___Werror" = xyes; then :
-  CFLAGS="${CFLAGS} -Werror"
+#  CFLAGS="${CFLAGS} -Werror"
       OCFLAGS="${OCFLAGS} -Werror"
 else
   as_CACHEVAR=`$as_echo "ax_cv_check_cflags__${SUNCC_CFLAGS}" | $as_tr_sh`
