diff -rup varnish-5.0.0.orig/bin/varnishd/http1/cache_http1_vfp.c varnish-5.0.0/bin/varnishd/http1/cache_http1_vfp.c
--- varnish-5.0.0.orig/bin/varnishd/http1/cache_http1_vfp.c	2017-08-02 22:59:27.669331942 +0200
+++ varnish-5.0.0/bin/varnishd/http1/cache_http1_vfp.c	2017-08-02 22:59:59.870371241 +0200
@@ -155,7 +155,7 @@ v1f_pull_chunked(struct vfp_ctx *vc, str
 		if (q == NULL || *q != '\0')
 			return (VFP_Error(vc, "chunked header number syntax"));
 		cl = (ssize_t)cll;
-		if((uintmax_t)cl != cll)
+		if(cl < 0 || (uintmax_t)cl != cll)
 			return (VFP_Error(vc, "bogusly large chunk size"));
 
 		vfe->priv2 = cl;
