--- bin/varnishtest/tests/v00006.vtc.orig	2010-10-26 23:41:00.105106948 +0200
+++ bin/varnishtest/tests/v00006.vtc	2010-10-26 23:42:42.727231982 +0200
@@ -10,7 +10,7 @@
 } -start
 
 # Only one pool, to avoid getting more than one work thread
-varnish v1 -arg "-p thread_pools=1 -w1,1,300" -vcl {
+varnish v1 -arg "-p thread_pools=1 -w1,1,300 -s file,/var/tmp,1G" -vcl {
 	backend b1 {
 		.host = "127.0.0.1";
 		.port = "9080";
